scripts/distribution/macOS-package/NodeConfigurationInstallerPlugin/NodeConfigurationInstallerPlugin/MyInstallerPane.m

```diff
@@ -139,37 +142,70 @@ - (void) saveConfigurationToDisk {
     }
 }
 
-// Checks that both the mainnet and testnet node names:
-// - have length > 0 and <= 100
-// - only contain characters in [a-zA-Z0-9-_ ]
+// Checks that a node name:
+// - has length > 0 and <= 100 if at least one of `run on startup`, `run after install`, or `report to network dashboard` is checked
+// - only contains characters in [a-zA-Z0-9-_ ]
 // - cannot start or end with spaces
-- (BOOL) nodeNamesAreValid
+- (BOOL) nodeNameIsValid:(NSTextField*)nodeNameField
 {
+    // If the node name field is disabled no  need to check the name validity
+    if (!nodeNameField.isEnabled) {
+        return TRUE;
+    }
+
+    NSString *nodeName = [nodeNameField stringValue];
+
     // NB: This character set must be accepted by the service file XML parser.
     // This naturally excludes '<', '>', etc., but the service won't start with a sequence of ',.' in the node name either.
     NSCharacterSet *allowedChars = [[NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_ "] invertedSet];
 
     // Default maximum node name length that the collector-backend will accept. Measured in UTF-8 encoded bytes.
     int maxLen = 100;
 
-    NSString *mainnetName = [_oMainnetNodeName stringValue];
-    NSString *testnetName = [_oTestnetNodeName stringValue];
-
     // NB: Get the actual length in bytes if allowedChars is changed to contain non-ascii characters.
-    NSUInteger mainnetNameLen = [mainnetName length];
-    NSUInteger testnetNameLen = [mainnetName length];
-
-    return      mainnetNameLen > 0
-            &&  mainnetNameLen <= maxLen
-            &&  testnetNameLen > 0
-            &&  testnetNameLen <= maxLen
-            &&  ![mainnetName hasPrefix:@" "]
-            &&  ![mainnetName hasSuffix:@" "]
-            && !([mainnetName rangeOfCharacterFromSet:allowedChars].location != NSNotFound)
-            && !([testnetName rangeOfCharacterFromSet:allowedChars].location != NSNotFound);
+    NSUInteger nodeNameLen = [nodeName length];
+
+    return      nodeNameLen > 0
+            &&  nodeNameLen <= maxLen
+            &&  ![nodeName hasPrefix:@" "]
+            &&  ![nodeName hasSuffix:@" "]
+            && !([nodeName rangeOfCharacterFromSet:allowedChars].location != NSNotFound);
 }
 
-@end
+// Checks that both the mainnet and testnet node names are valid.
+- (BOOL) nodeNamesAreValid
+{
+    BOOL mainnetNameIsValid = [self nodeNameIsValid:_oMainnetNodeName];
+    BOOL testnetNameIsValid = [self nodeNameIsValid:_oTestnetNodeName];
 
+    return mainnetNameIsValid && testnetNameIsValid;
```

Comment:
If you deselect all options, then this will also be `true`.
I am not sure whether this is a feature or a bug because I am not sure about the value you would get from it.
What do you think?

Reply:
Good catch. I am not sure either. But let's prevent leaving both empty.  I'll push another commit.

Reply:
done